apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NsxTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
+}
+
+type NsxT struct {
+	// A flag that enables pod routing
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	PodRoutingEnabled bool      `json:"podRoutingEnabled"`
+	Routes            NsxTRoute `json:"routes"`
+
+	// The username used to access NSX-T
+	Username string `json:"username"`
+
+	// The password used to access NSX-T
+	Password string `json:"password"`
```

Comment:
password needs to be a ref to a secret

Reply:
changed to use secret ref for vsphere & nsxt credentials